plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -229,6 +231,29 @@ public void testV2TableWithEqualityDeleteWhenColumnIsNested()
         assertQuery("SELECT array_column[1], map_column[1], row_column.x FROM " + tableName, "SELECT 1, 2, 1 FROM nation WHERE regionkey != 1");
     }
 
+    @Test
+    public void testParquetGeneratedByPyArrow()
+    {
+        String tableName = "test_parquet_by_pyarrow" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + "(names ARRAY(varchar))");
```

Comment:
Is the problem specific to complex types ?
I got the impression that it could happen for any primitive type when field ids are not in parquet footer schema

Reply:
It's specific to compelx types. I initially tested with primitive types, but it doesn't cause a query failure.